packages/doenetml-worker-rust/lib-doenetml-core/src/core/state/types/list_marker.rs

```diff
@@ -0,0 +1,214 @@
+    pub fn index_to_formatted_string(&self, index: i64) -> String {
+        let marker = self.index_to_string(index);
+        let needs_decimal = match self {
+            ListMarker::Circle | ListMarker::Square | ListMarker::Disc => false,
+            _ => {
+                // All other types are for `<ol>` lists.
+                true
+            }
+        };
+
+        if needs_decimal {
+            format!("{}.", marker)
+        } else {
+            marker
+        }
+    }
+
+    /// The default value of an unordered list.
+    pub fn default_ul() -> Self {
+        Self::Disc
+    }
+    /// The default value of an ordered list.
+    pub fn default_ol() -> Self {
+        Self::Decimal { start: 1 }
+    }
+    /// Whether this list marker is ordered.
+    pub fn is_ordered(&self) -> bool {
+        matches!(
+            self,
+            ListMarker::Decimal { .. }
+                | ListMarker::LowerAlpha
+                | ListMarker::LowerRoman
+                | ListMarker::UpperAlpha
+                | ListMarker::UpperRoman
+        )
+    }
+}
+
+impl Default for ListMarker {
+    fn default() -> Self {
+        Self::Decimal { start: 1 }
+    }
+}
+
+impl From<&str> for ListMarker {
+    fn from(s: &str) -> Self {
+        match s {
+            "1" => Self::Decimal { start: 1 },
+            "0" => Self::Decimal { start: 0 },
+            "a" => Self::LowerAlpha,
+            "i" => Self::LowerRoman,
+            "A" => Self::UpperAlpha,
+            "I" => Self::UpperRoman,
+            "decimal" => Self::Decimal { start: 1 },
+            "lower-alpha" => Self::LowerAlpha,
+            "lower-roman" => Self::LowerRoman,
+            "upper-alpha" => Self::UpperAlpha,
+            "upper-roman" => Self::UpperRoman,
+            "disc" => Self::Disc,
+            "circle" => Self::Circle,
+            "square" => Self::Square,
+            _ => Self::Decimal { start: 1 },
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_list_marker_from_str() {
+        assert_eq!(ListMarker::from("1"), ListMarker::Decimal { start: 1 },);
+        assert_eq!(ListMarker::from("0"), ListMarker::Decimal { start: 0 });
+        assert_eq!(ListMarker::from("a"), ListMarker::LowerAlpha);
+        assert_eq!(
+            ListMarker::from("INVALID"),
+            ListMarker::Decimal { start: 1 }
+        );
+    }
+
+    #[test]
+    fn test_list_marker_next() {
+        assert_eq!(
+            ListMarker::Decimal { start: 1 }.next_marker(),
+            ListMarker::LowerAlpha
+        );
+        assert_eq!(ListMarker::LowerAlpha.next_marker(), ListMarker::LowerRoman);
+        assert_eq!(ListMarker::LowerRoman.next_marker(), ListMarker::UpperAlpha);
+        assert_eq!(ListMarker::UpperAlpha.next_marker(), ListMarker::UpperRoman);
+    }
+
+    #[test]
+    fn can_convert_to_strings() {
+        assert_eq!(ListMarker::Decimal { start: 1 }.index_to_string(1), "1");
```

Comment:
I suspect you'll also get "1" if you use another value for `start`.

Reply:
The line right below uses another value. The `1` is completely ignored when you call `index_to_string`.